### ISSUE TITLE
CONTRIBUTING.md should contain a definition of done for components

### ISSUE THREAD

**Issue by nyaray** [scenic-contrib/scenic-widget-contrib#2]
Discussion thread for discussing when a component is "done".

**Comment by nyaray** [comment]
I think the component should
- [ ] be well documented, with examples
- [ ] implement the info callback properly
- [ ] handle theming properly

Not sure:
- [ ] how/if there should be tests

What else do you think should be mandatory?

**Comment by lawik** [comment]
A table or otherwise clear definition of what version works with what version of Scenic maybe?
I mean you could check mix.exs for every release to find what works for your project. But from having worked with web UI frameworks on legacy framework versions, any component that can tell me in the README or docs how far back I'd need to go to get it working is doing something right.

**Comment by nyaray** [comment]
I'm not sure that I'd like to add "backwards compatibility checking for old-ass versions of scenic" to a list somewhere. I might be missing your point, but I'm not sure it's a good idea to have arbitrary backwards compatibility as a goal.

EDIT: maybe if you can work it into a CI pipeline somehow, though... :)

**Comment by lawik** [comment]
No-no, the requirement would be much simpler. When you release a new version, note in the table in your docs what version it works with. This way someone looking for the right version for them can just see "Ah, I'm on good old 0.10, then I SuperComponent v1.3. One day I too will use Scenic 1.5 with SuperComponent v5.8 but it is not today. For I maintain legacy."

I don't care that they update the old version to verify whether it would work with newer framework versions, just that we know what are known good versions. I know that can be quite useful at least.

**Comment by nyaray** [comment]
so... pretty much a line in the `@moduledoc` of the component, stating which the earliest tested Scenic version is?

**Comment by lawik** [comment]
That wasn't exactly what I was thinking. It could work if the module uses hex docs and the module doc becomes documentation in a nice way.
I was mostly thinking some kind of reference in the README.md. But it might be unnecessary. Since what I'm thinking of is mostly two purposes:

- Identify stale components that you might not be able to use with your current version of Scenic.
- Identify which component version you can use with a stale version of Scenic.

And if we have all the component repos in a single org, creating tables for these things could be entirely automated. Though I still think it might be a nice thing to have, maybe it is an unnecessary requirement.
